File: Source/Samples/95_Periodic/code_awidget.cpp
// AWidget application source code
#include "PeriodicApp.h"

void PeriodicApp::setup_awidget( AWidget *layout )
{
    PODVector<AWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, URHO3D_HANDLER(PeriodicApp, HandleAwidgetEvent ));
}

void PeriodicApp::HandleAwidgetEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    AWidget* widget = static_cast<AWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uiwidgetcode" )
        {
            AppLog( "AWidget support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Sources/code_awidget.cpp", widget->GetParent() );
        }
        if (widget->GetId() ==  "uiwidgetlayout" )
        {
            AppLog( "AWidget support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_awidget.ui.txt", widget->GetParent() );
        }
    }
}

File: bin/Data/AUI/Periodic/LuaScripts/code_awidget.lua
-- AWidget application source code


function setup_uiwidget( layout )
    local dest = layout:SearchAWidgetClass( "TBButton" )
    for i, awid in ipairs(dest) do 
        SubscribeToEvent(awid, "WidgetEvent", "HandleUiwidgetEvent")
    end
end

function HandleUiwidgetEvent(eventType, eventData)
    local widget = eventData["Target"]:GetPtr("AWidget")
    if widget == nil then
        return
    end
    if eventData["Type"]:GetInt() == UI_EVENT_TYPE_CLICK then
        if widget:GetId() == "uiwidgetcode" then
            AppLog( "AWidget support : " .. widget:GetId() .. " was pressed " )
            ViewCode ( "LuaScripts/code_awidget.lua", widget:GetParent())
        end
        if widget:GetId() ==  "uiwidgetlayout" then
            AppLog( "AWidget support : " .. widget:GetId() .. " was pressed ")
            ViewCode ( "Scenes/layout_awidget.ui.txt", widget:GetParent())
        end
    end
end

